lib/nexpose/wait.rb

```diff
@@ -0,0 +1,133 @@
+module Nexpose
+
+  class Wait
+    attr_reader :error_message, :ready, :retry_count, :timeout, :polling_interval
+
+
+    def initialize(retry_count: nil, timeout: nil, polling_interval: nil)
```

Comment:
I think maybe it'd be better to convert Wait into a Module instead of having it as Class. Then instead of instantiating an instance of it for use we could just call the for_ methods and pass in a poller if we wanted that kind of granularity to the wait.  Otherwise it would just create a Poller with default settings and use that instead.  This way we wouldn't have to do

    report_wait = Nexpose::Wait.new(timeout: 60, retry: 2)
    report_wait.for_report( nsc: @console, report_id: config.id )

since this would require the variable name to make sense when invoking the for_ methods
Instead something like

    report_wait = Nexpose::Wait.for_report( nsc: @console, report_id: config.id )

and if they wanted to set their own custom timeout they would then define a poller and pass it in.

    poller = Nexpose::Poller.new(timeout: @timeout, polling_interval: @polling_interval)
    report_wait = Nexpose::Wait.for_report(nsc: @console, report_id: config.id, poller: poller)

the for_ methods would return a map with the state and message of the waiting call